src/UHHNtupleConverterModule.cxx

```diff
@@ -488,165 +490,91 @@ UHHNtupleConverterModule::UHHNtupleConverterModule(Context & ctx){
-        JEC_AK8chs_F = JERFiles::Fall17_17Nov2017_V32_F_L123_AK8PFchs_DATA;
-
-        JEC_AK4puppi_B = JERFiles::Fall17_17Nov2017_V32_B_L123_AK4PFPuppi_DATA;
-        JEC_AK4puppi_C = JERFiles::Fall17_17Nov2017_V32_C_L123_AK4PFPuppi_DATA;
-        JEC_AK4puppi_D = JERFiles::Fall17_17Nov2017_V32_D_L123_AK4PFPuppi_DATA;
-        JEC_AK4puppi_E = JERFiles::Fall17_17Nov2017_V32_E_L123_AK4PFPuppi_DATA;
-        JEC_AK4puppi_F = JERFiles::Fall17_17Nov2017_V32_F_L123_AK4PFPuppi_DATA;
-
-	JEC_AK8puppi_B = JERFiles::Fall17_17Nov2017_V32_B_L123_AK8PFPuppi_DATA;
-	JEC_AK8puppi_C = JERFiles::Fall17_17Nov2017_V32_C_L123_AK8PFPuppi_DATA;
-	JEC_AK8puppi_D = JERFiles::Fall17_17Nov2017_V32_D_L123_AK8PFPuppi_DATA;
-	JEC_AK8puppi_E = JERFiles::Fall17_17Nov2017_V32_E_L123_AK8PFPuppi_DATA;
-	JEC_AK8puppi_F = JERFiles::Fall17_17Nov2017_V32_F_L123_AK8PFPuppi_DATA;
+	jec_tag = "Fall17_17Nov2017";
+	jec_ver = "32";
+	std::cout << "USING 2017 MC JEC: "<< jec_tag << " V" << jec_ver << std::endl;
       }
       else if(year == Year::is2018 ){
-	std::cout << "USING 2016 MC JEC" << std::endl;
-        JEC_AK8chs_A = JERFiles::Autumn18_V8_A_L123_AK8PFchs_DATA;
-	JEC_AK8chs_B = JERFiles::Autumn18_V8_B_L123_AK8PFchs_DATA;
-	JEC_AK8chs_C = JERFiles::Autumn18_V8_C_L123_AK8PFchs_DATA;
-	JEC_AK8chs_D = JERFiles::Autumn18_V8_D_L123_AK8PFchs_DATA;
-
-        JEC_AK4puppi_A = JERFiles::Autumn18_V8_A_L123_AK4PFPuppi_DATA;
-	JEC_AK4puppi_B = JERFiles::Autumn18_V8_B_L123_AK4PFPuppi_DATA;
-	JEC_AK4puppi_C = JERFiles::Autumn18_V8_C_L123_AK4PFPuppi_DATA;
-	JEC_AK4puppi_D = JERFiles::Autumn18_V8_D_L123_AK4PFPuppi_DATA;
-
-	JEC_AK8puppi_A = JERFiles::Autumn18_V8_A_L123_AK8PFPuppi_DATA;
-	JEC_AK8puppi_B = JERFiles::Autumn18_V8_B_L123_AK8PFPuppi_DATA;
-	JEC_AK8puppi_C = JERFiles::Autumn18_V8_C_L123_AK8PFPuppi_DATA;
-	JEC_AK8puppi_D = JERFiles::Autumn18_V8_D_L123_AK8PFPuppi_DATA;
+	jec_tag = "Autumn18";
+	jec_ver = "8";
+	std::cout << "USING 2018 MC JEC: "<< jec_tag << " V" << jec_ver << std::endl;
       }
-      for (auto it = JEC_AK8chs_B.begin(), end = JEC_AK8chs_B.end(); it != end; ++it){ std::cout <<"AK8chs B JEC: " << *it << std::endl;}
-    }
-
-
-    if(isMC){
-      jet_corrector.reset(new JetCorrector(ctx, JEC_AK8chs));
-      jet_corrector_puppi.reset(new GenericJetCorrector(ctx, JEC_AK4puppi,"jetsAk4Puppi"));
-      topjet_corrector.reset(new TopJetCorrector(ctx, JEC_AK8puppi));
+      std::cout << "for the following jet collections: " << jec_jet_coll_AK8chs << " " << jec_jet_coll_AK4puppi << " " << jec_jet_coll_AK8puppi << std::endl;     
+      jet_corrector.reset(new JetCorrector(ctx, JERFiles::JECFilesMC(jec_tag, jec_ver, jec_jet_coll_AK8chs)));
+      jet_corrector_puppi.reset(new GenericJetCorrector(ctx, JERFiles::JECFilesMC(jec_tag, jec_ver, jec_jet_coll_AK4puppi),"jetsAk4Puppi"));
+      topjet_corrector.reset(new TopJetCorrector(ctx, JERFiles::JECFilesMC(jec_tag, jec_ver, jec_jet_coll_AK8puppi)));
     }
```

Comment:
We are using three different jets/top collections:
<Item Name="JetCollection" Value="jetsAk8CHS" />
<Item Name="TopJetCollection" Value="jetsAk8PuppiSubstructure_SoftDropPuppi"/>
<Item Name="additionalBranches" Value="jetsAk4Puppi" /> <!-- additional jet collection for VBF jets -->

And if I am not mistaken, here I am assigning the correspondent corrections.

Reply:
True! However, the topjet collection is puppi-softdrop jets which should not be corrected (also we do not use the 4-vectors anywhere). While we apply the puppi-softdrop mass correction on the mass separately. We should be sure that we are not applying anything on top?!

Reply:
True! I did it already wrong before. Then I should remove the "process" for topjetcollector https://github.com/Diboson3D/UHHNtupleConverter/blob/master/src/UHHNtupleConverterModule.cxx#L812